processing/src/main/java/io/druid/segment/IndexMergerV9.java

```diff
@@ -335,7 +335,8 @@ private void makeDimensionColumns(
 
     long startTime = System.currentTimeMillis();
     final BitmapSerdeFactory bitmapSerdeFactory = indexSpec.getBitmapSerdeFactory();
-    final CompressedObjectStrategy.CompressionStrategy compressionStrategy = indexSpec.getDimensionCompressionStrategy();
+    final boolean compressed =
```

Comment:
@xvrl It's related to what you've mentioned above (keeping changes just in new V9 merger). It writes `version` value in header properly and backward compatible.